functions/api.php

```diff
@@ -102,15 +102,19 @@ private static function prepare_external_story_for_response( $post, $request ) {
                 'description'  => '',
                 'url'          => '',
                 'source'       => '',
+                'source_name'  => '',
+                'source_image' => '',
                 'publish_date' => '',
                 'categories'   => array()
             );
 
             $retval['title']        = $post->post_title;
             $retval['link_text']    = get_post_meta( $post->ID, 'externalstory_text', true );
             $retval['description']  = get_post_meta( $post->ID, 'externalstory_description', true );
-            $retval['url']          = get_post_meta( $post->ID, 'externalstory_url', true );
-            $retval['source']       = get_post_meta( $post->ID, 'externalstory_source', true );
+			$retval['url']          = get_post_meta( $post->ID, 'externalstory_url', true );
+			$retval['source']       = get_post_meta( $post->ID, 'externalstory_source', true );
+			$retval['source_name']  = wp_get_post_terms( $post->ID, 'sources' )[0]->name;
+            $retval['source_image'] = get_field( 'news_source_image', wp_get_post_terms( $post->ID, 'sources' )[0] )['url'];
```

Comment:
line 117:  `wp_get_post_terms( $post->ID, 'sources' )[0]` will throw at least a notice if `wp_get_post_terms()` doesn't return at least one item, or if it returns `WP_Error`.

Trying to access `get_field(...)['url']` may also throw similar notices if a valid value isn't returned.